enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/interceptor/MediationPolicyFilter.java

```diff
@@ -32,51 +39,76 @@
             for (Policy policy : policyConfig.getIn()) {
-                applyPolicy(requestContext, policy);
+                if (!applyPolicy(requestContext, policy)) {
+                    return false;
+                }
             }
         }
         return true;
     }
 
-    private void applyPolicy(RequestContext requestContext, Policy policy) {
-        //todo(amali) check policy order
-        switch (policy.getAction()) {
-            case "SET_HEADER": {
-                addOrModifyHeader(requestContext, policy.getParameters());
-                break;
-            }
-            case "RENAME_HEADER": {
-                renameHeader(requestContext, policy.getParameters());
-                break;
-            }
-            case "REMOVE_HEADER": {
-                removeHeader(requestContext, policy.getParameters());
-                break;
-            }
-            case "ADD_QUERY": {
-                addOrModifyQuery(requestContext, policy.getParameters());
-                break;
-            }
-            case "REMOVE_QUERY": {
-                removeQuery(requestContext, policy.getParameters());
-                break;
-            }
-            case "REWRITE_RESOURCE_PATH": {
-                removeAllQueries(requestContext, policy.getParameters());
-                break;
-            }
-            case "REWRITE_RESOURCE_METHOD": {
-                modifyMethod(requestContext, policy.getParameters());
-                break;
+    private boolean applyPolicy(RequestContext requestContext, Policy policy) {
+        try {
+            switch (policy.getAction()) {
+                case "SET_HEADER": {
+                    addOrModifyHeader(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "RENAME_HEADER": {
+                    renameHeader(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "REMOVE_HEADER": {
+                    removeHeader(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "ADD_QUERY": {
+                    addOrModifyQuery(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "REMOVE_QUERY": {
+                    removeQuery(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "REWRITE_RESOURCE_PATH": {
+                    removeAllQueries(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "REWRITE_RESOURCE_METHOD": {
+                    modifyMethod(requestContext, policy.getParameters());
+                    return true;
+                }
+                case "OPA": {
+                    return opaAuthValidation(requestContext, policy.getParameters());
+                }
```

Comment:
Just to clarify: 
Can we add a default case (to handle unsupported actions apart from the null pointer situations) ? Or are we going to handle that scenario after line number 107 ?

Reply:
yes this is handled in line 107